packages/phalanx/test/Field/Field.cpp

```diff
@@ -354,6 +354,18 @@ TEUCHOS_UNIT_TEST(field, all)
       c_f7.setFieldData(PHX::KokkosViewFactory<double,typename PHX::DevLayout<double>::type,PHX::Device>::buildView(c_f7.fieldTag()));
     }
 
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Kokkos cast DynRankView accessor
+    {
+      auto kva = a.get_view();
+      Kokkos::deep_copy(kva, a.get_view());
+
+      auto kva_s = a.get_static_view();
+      Kokkos::deep_copy(kva_s, a.get_view());
+
+      auto kvc = c.get_view();
+      Kokkos::deep_copy(kvc, c.get_static_view());
+    }
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
@bathmatt - Looks like this test code was added to the wrong test. There are two versions of field objects: "PHX::MDField" implemented in Phalanx_MDField.hpp and a "PHX::Field" implemented in Phalanx_Field.hpp. For MDField, you would have to add the test code to two test files MDField_Runtime.cpp and MDField_Compiletime.cpp. The runtime and compile time versions mirror each other but use different code paths underneath depending on whether the underlying kokkos view has a static or dynamic rank. The Field.cpp file is for testing the PHX::Field object. In case you are wondering, the MDField was the original object and allows for the extent information from structs (MDField<double,CELL,IP,DIM>), while the Field object takes an integer for the rank (Field<double,3>). Fields was requested by the empire fluids team.

Reply:
OK, Ill add it there and remove my other test.  Sorry, not up to speed on the testing here.